File: server/src/main/java/com/smartlock/server/lock/persistence/model/Lock.java
package com.smartlock.server.lock.persistence.model;

import com.smartlock.server.lock.presentation.dto.CreateLockDto;
import org.springframework.lang.Nullable;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity(name="lockEntity")
public class Lock {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;
    private String uuid;
    @Nullable
    private String name;
    private boolean active;
    private boolean isOpen;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getUuid() {
        return uuid;
    }

    public void setUuid(String uuid) {
        this.uuid = uuid;
    }

    public boolean isActive() {
        return active;
    }

    public void setActive(boolean active) {
        this.active = active;
    }

    public boolean isOpen() {
        return isOpen;
    }

    public void setOpen(boolean open) {
        isOpen = open;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Lock(CreateLockDto createLockDto) {
        this.uuid = createLockDto.getUuid();
        this.active = true;
        this.isOpen = true;
        this.name = createLockDto.getName();
    }

    public Lock() {
        this.isOpen = true;
    }
}

File: lock_app/lib/data/lock/lock.dart
class Lock {
  final String name;
  final int id;
  final String uuid;
//  final int userAdminId;
//  bool isActive;

  Lock(this.name, this.id, this.uuid);

  Map dataToMap() {
    Map<String, dynamic> map = {
      "name": name,
      "id": id,
      "uuid": uuid,
//      "userAdminId": userAdminId,
//      "isActive": isActive
    };
    return map;
  }

  Lock.fromMap(Map map)
      : name = map['name'],
        id = map['id'],
        uuid = map['uuid'];
//        userAdminId = map['userAdminId'],
//        isActive = map['isActive'];
}
